src/Toml/Printer.hs

```diff
@@ -171,13 +173,17 @@ prettyTableArrays options i pref = mapOrdered arrText options
 
 -- Returns an indentation prefix
 tabWith :: PrintOptions -> Int -> Text
-tabWith options n = Text.replicate (n * indent options) " "
+tabWith options n = Text.replicate (n * printOptionsIndent options) " "
 
 -- Returns a proper sorting function
-mapOrdered :: Ord k => ((k, v) -> [t]) -> PrintOptions -> HashMap k v -> [t]
-mapOrdered f options
-    | shouldSort options = concatMap f . sortOn fst . HashMap.toList
-    | otherwise          = concatMap f . HashMap.toList
+mapOrdered :: ((Key, v) -> [t]) -> PrintOptions -> HashMap Key v -> [t]
+mapOrdered f options = case printOptionsSorting options of
+    Just sorter -> concatMap f . sortBy (applyToFirst sorter) . HashMap.toList
```

Comment:
You don't need to write `applyToFirst`, you can use standard function `comparing` here:

* https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Ord.html#v:comparing

Reply:
Perhaps I'm missing something? `comparing` doesn't accept a user-provided `a -> a -> Ordering` sorter. I couldn't find anything in the standard library that applies a custom sorter to tuples, either. (I attempted at some length to discover some slick `Functor` or `Applicative` thing to "lift" the user-provided sorter to tuples, but I failed and figured this one-line helper would be more readable anyway.)

Reply:
@ramanshah ah, sorry, that was an oversight from my side. Here you indeed can't use `comparing`. I also don't see any elegant implementation using some standard functions. In that case, I can only recommend changing the type signature of the `applyToFirst` function further using _generalisation programming pattern_ in Haskell to the following (so there is only one possible implementation without any chance to introduce error):

```haskell
applyToFirst :: (a -> b -> c) -> (a, x) -> (b, y) -> c
```

Reply:
Very cool idea. I'm learning a lot on in this PR :)